lib/rules/no-unused-expressions.js

```diff
@@ -50,6 +50,10 @@ module.exports = {
                     allowTaggedTemplates: {
                         type: "boolean",
                         default: false
+                    },
+                    disallowJSX: {
```

Comment:
I see how the name `disallowJSX` follows from the implementation, but I'm wondering if the double negative might confuse users. For consistency with the other options, I could go for your original option name `allowJSX` but defaulting to `true` so it's not a breaking change. I also think `checkJSX` defaulting to `false` would be clear. Anyone have any preferences?

Reply:
It’s always super weird when absence and false aren’t the same - iow, i think a boolean option should never default to true.

Reply:
That’s a great point. I also wasn’t thrilled about another “allow” option having the opposite default, so `checkJSX` seems like the better trade off now.

Reply:
Maybe `enforceForJSX`?